### PR TITLE
Transfuser

### PR DESCRIPTION
**Pull request by ign-saurav**
### Overview
This PR integrates Transfuser model support into the Tenstorrent codebase(`tt-metal`), enabling transformer based multimodel sensor fusion on TTNN devices .

Input Resolutions

- Image (RGB, NCHW): 1 × 3 × 160 × 704

   - Three front cameras are cropped & concatenated to 160×704.

- LiDAR (BEV, NCHW): 1 × 1 × 256 × 256

  - BEV grid input at 256×256.

### Source
- [Transfuser](https://github.com/autonomousvision/transfuser)

### TransFuser – Model Tree

transfuser
├── image_encoder/ # RegNet (RGB: 1×3×160×704)
│ ├── stem
│ └── stages s1–s4 # RegNet bottleneck blocks
│
├── lidar_encoder/ # RegNet (BEV: 1×1×256×256)
│ ├── stem
│ └── stages s1–s4
│
├── projection/
│ ├── img_proj # to transformer dim D
│ └── lidar_proj
│
├── fusion_transformer/
│ ├── encoder_blocks (L) # MHSA + MLP
│ └── cross_attention # img ↔ lidar
│
├── fusion_head/
│ └── upsample_path # upsample + (depthwise) conv
│
└── heads/
├── semantic_head # segmentation logits
├── center_head # instance center heatmap
└── offset_head # instance offsets

#### Hardware: Wormhole n150
seed=42

| Backbone Output | PCC Score |
|-----------------|-----------|
| p2 (FPN)        | 0.9995 |
| p3 (FPN)        | 0.9996 |
| p4 (FPN)        | 0.9994 |
| p5 (FPN)        | 0.9992 |
| Image Grid      | 0.9993 |
| Fused Features  | 0.9950 |

### Per-Head PCC Breakdown

| Component                | PCC Score |
|--------------------------|-----------|
| Fused Features           | 0.9947 |
| Predicted Waypoints      | 0.9999 |
| Feature Map              | 0.9997 |
| Detection Head (Overall) | 0.9973 |
| Bounding Boxes (Boxes)   | 0.9379 |

### Detection Head – Per-Head PCC Breakdown

| Head Component | PCC Score |
|---------------|-----------|
| Heatmap       | 0.9973 |
| WH (Width/Height) | 0.9992 |
| Offset        | 0.9979 |
| Yaw Class     | 0.9997 |
| Yaw Residual  | 0.9932 |
| Velocity      | 0.9993 |
| Brake         | 0.9998 |

### Tests
Full network test (with real weights and real input from image and lidar):
`pytest models/experimental/transfuser/tests/test_lidar_center_net.py`

### Demo Test:
`python models/experimental/transfuser/demo/lidar_center_net_demo.py   --data-root <data_root_path>   --frame <frame_number>   --weights <checkpoint_path> --device-id 0`
Eg:
`python models/experimental/transfuser/demo/lidar_center_net_demo.py   --data-root Scenario3_Town01_curved_route0_11_23_20_02_59   --frame 0120   --weights model_seed1_39.pth --device-id 0`

### Performance:

Total device time : 47104us
FPS : 21.2

### Layers running on torch

- Other than transfuser rest of lidarcenternet is running on torch

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by ign-saurav** [comment]
@sankarmanoj-tt, Could you please review the PR?
